main.cpp

```diff
@@ -418,60 +491,76 @@ void FourBandPEQ::speakerCompensation()
     std::cout << "Your budget speakers have been optimized\n";
 }
 
+struct FourBandPEQWrapper
+{
+    FourBandPEQWrapper( FourBandPEQ* ptr ) : pointerToFourBandPEQ( ptr ) { }
+    ~FourBandPEQWrapper()
+    {
+        delete pointerToFourBandPEQ;
+        pointerToFourBandPEQ = nullptr;
+    }
+    
+    FourBandPEQ* pointerToFourBandPEQ = nullptr;
+};
+
 /*
  MAKE SURE YOU ARE NOT ON THE MASTER BRANCH
-
  Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
  
  If you didn't already:
     Make a pull request after you make your first commit
     pin the pull request link and this repl.it link to our DM thread in a single message.
-
  send me a DM to review your pull request when the project is ready for review.
-
  Wait for my code review.
  */
 
 #include <iostream>
 int main()
 {
-    Oscilloscope oScope;  //3) instatiation of UDT
-    oScope.measureVoltage(2.01f, 4.123f);
-    oScope.graphWaveformDifference(32, 16);
-    oScope.measurement(5.123f, 6.321f);
+    Oscilloscope* oScope = new Oscilloscope;
+    OscilloscopeWrapper oscilloscopeWrapper( oScope );
```

Comment:
Don't declare a local variable and pass it to your wrapper instance.
pass a new instance directly in the constructor:
```
OscilloscopeWrapper oscilloscopeWrapper( new Oscilloscope );
```
what you are doing is unsafe, because you can interact with `oScope` outside of the wrapper class, which is managing it.
- [ ] revise

Reply:
Thanks.  This does make more sense.